fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,198 @@
+     */
+    private boolean doesSchemaContainAllDimensionFields(Dimension dimension, String avroSchemaPath)
+        throws IllegalArgumentException {
+
+        // Convert the AVRO schema file into JsonNode Object
+        JsonNode jsonAvroSchema = convertFileToJsonNode(avroSchemaPath);
+
+        try {
+            // The following method throws a NullPointerException, if the JSON schema doesn't have the `fields` object
+            // Avoiding the `Optional.of().orElseThrow()` to log the error details.
+            Optional.of(jsonAvroSchema.get("fields"));
+        } catch (NullPointerException e) {
+            String msg = "`fields` is a required JSON field in the avro schema {} ";
+            LOG.error(msg, e);
+            throw new IllegalArgumentException(msg);
+        }
+
+        jsonAvroSchema = jsonAvroSchema.get("fields");
+
+        // Populating the set of avro field names
+        Set<String> avroFields = StreamSupport.stream(jsonAvroSchema.spliterator(), false)
+                .map(jsonNode -> jsonNode.get("name"))
+                .map(JsonNode::asText)
+                .collect(Collectors.toSet());
+
+        // True only if all of the mapped dimension fields are present in the Avro schema
+        return dimension.getDimensionFields().stream()
+                .map(dimensionField -> dimensionFieldNameMapper.convert(dimension, dimensionField))
+                .allMatch(avroFields::contains);
+    }
+
+    /**
+     * Parses the avro file and populates the dimension rows after validating the schema.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroFilePath The path of the AVRO data file (.avro)
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return A set of dimension rows
+     *
+     * @throws IllegalArgumentException thrown if JSON object `fields` is not present
+     */
+    public Set<DimensionRow> parseAvroFileDimensionRows(Dimension dimension, String avroFilePath, String avroSchemaPath)
+        throws IllegalArgumentException {
+
+        if (doesSchemaContainAllDimensionFields(dimension, avroSchemaPath)) {
+
+            DataFileReader<GenericRecord> dataFileReader;
+            try {
+                // Creates an AVRO schema object based on the parsed AVRO schema file (.avsc)
+                Schema schema = new Schema.Parser().parse(new File(avroSchemaPath));
+
+                // Creates an AVRO DatumReader object based on the AVRO schema object
+                DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(schema);
+
+                // Creates an AVRO DataFileReader object that reads the AVRO data file one record at a time
+                dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader);
+
+            } catch (IOException e) {
+                String msg = String.format("Unable to process the file, at the location %s {} ", avroSchemaPath);
```

Comment:
`{}` not needed, since the logging framework isn't doing any additional expansion.